### PR TITLE
Fixes to ExitStatus and its docs

### PR DESCRIPTION
**Pull request by ijackson**
* On Unix, properly display every possible wait status (and don't panic on weird values)
* In the documentation, be clear and consistent about "exit status" vs "wait status".

### CODE REVIEW

**Review by dtolnay — COMMENTED** [review]
LGTM once the tidy failure is resolved. Thanks!

**Comment by ijackson** [comment]
> LGTM once the tidy failure is resolved. Thanks!

Oops. My privsep workflow means I sometimes forget the tidy check.  Fixed now (and squashed the fix in).  Thanks.

**Comment by ijackson** [comment]
@dtolnay Hi.  Sorry to hassle you personally, but is there anything else I need to do here?  I have a branch for #73125 which is blocked on this.

**Review by dtolnay — APPROVED** [review]
Thanks for the reminder. I just had way too many things stack up, sorry for letting this drop. Looks good!

**Comment by ijackson** [comment]
> Thanks for the reminder. I just had way too many things stack up, sorry for letting this drop. Looks good!

No problem, I know the feeling.  Thank you :-).

**Comment by JohnTitor** [comment]
Failed in rollup: https://github.com/rust-lang/rust/pull/82749#issuecomment-790525956
@bors r-

**Comment by ijackson** [comment]
> Failed in rollup: [#82749 (comment)](https://github.com/rust-lang/rust/pull/82749#issuecomment-790525956)

Sorry about that.  I will restrict the failing test cases to Linux.

**Comment by ijackson** [comment]
OK I think I have fixed this now.

@rustbot modify labels -S-waiting-on-author +S-waiting-on-review

**Comment by ijackson** [comment]
@JohnTitor Should I be asking you to re-approve it, or ...?   (Thanks for your attention and sorry for messing up your rollup.)

**Comment by ijackson** [comment]
@JohnTitor (or anyone) Also, if I may ask: if I have a branch which I suspect might have portability problems, is there a less disruptive way for me to get the full rust CI run on it than to try to get it approved, and wait for it to make a mess for you?

I could ask the reviewer to mark it not for rollup, but ideally I would get the computers to find the problems before bothering a human reviewer with re-review over portability fixes.

**Comment by JohnTitor** [comment]
Seems reasonable to me but I'm not part of the reviewers so r? @dtolnay 
Btw, don't worry about the rollup failure, it's not so hard to re-create a new one :)

> Also, if I may ask: if I have a branch which I suspect might have portability problems, is there a less disruptive way for me to get the full rust CI run on it than to try to get it approved, and wait for it to make a mess for you?

If it makes sense, we can run a try build (via `bors try`) to check the change on some more builders than PR's.

File: library/std/src/sys/unix/process/process_unix/tests.rs
Comment (dtolnay) [inline library/std/src/sys/unix/process/process_unix/tests.rs:22]: For tests that we only want to *run* on some platforms, but are capable of *compiling* perfectly fine on any platform, I would prefer to do the following instead. This makes things easier for someone who needs to refactor this code on a non-Linux host later. Completely conditional-compiling out the code would hide type errors or missing arguments and it commonly leads to needless round-trips through bors to get things right, which is a frustrating experience.

Using #[cfg] would typically only be if the compilation itself is platform-specific such as one of the std::os extension traits, or if for some reason there is immense cost in compiling the tests (which is not the case here).

```suggestion
    if cfg!(target_os = "linux") {
        t(0x0137f, "stopped (not terminated) by signal: 19");
        t(0x0ffff, "continued (WIFCONTINUED)");
    }
```

File: library/std/src/sys/unix/process/process_unix/tests.rs
Comment (dtolnay) [inline library/std/src/sys/unix/process/process_unix/tests.rs:28]:
```suggestion
    if cfg!(all(target_os = "linux", target_env = "gnu")) {
        t(0x000ff, "unrecognised wait status: 255 0xff");
    }
```

**Review by dtolnay — CHANGES_REQUESTED** [review]

File: library/std/src/sys/unix/process/process_unix/tests.rs
Comment (ijackson) [reply library/std/src/sys/unix/process/process_unix/tests.rs:22]: > For tests that we only want to _run_ on some platforms, but are capable of _compiling_ perfectly fine on any platform, I would prefer to do the following instead. This makes things easier for someone who needs to refactor this code on a non-Linux host later. Completely conditional-compiling out the code would hide type errors or missing arguments and it commonly leads to needless round-trips through bors to get things right, which is a frustrating experience.

Thanks.  That is a very good point.

**Review by dtolnay — APPROVED** [review]

**Pull request merged**
